irohad/ordering/impl/on_demand_os_client_grpc.cpp

```diff
@@ -64,16 +66,28 @@ OnDemandOsClientGrpc::onRequestProposal(consensus::Round round) {
   if (not response.has_proposal()) {
     return boost::none;
   }
-  return ProposalType{std::make_unique<shared_model::proto::Proposal>(
-      std::move(response.proposal()))};
+  return proposal_factory_->build(response.proposal())
+      .match(
+          [&](iroha::expected::Value<
+              std::unique_ptr<shared_model::interface::Proposal>> &v)
+              -> boost::optional<OdOsNotification::ProposalType> {
+            return ProposalType{std::move(v).value};
```

Comment:
Are the parentheses in the correct order? Not `std::move(v.value)`?

Reply:
Yes
The reference material is http://oliora.github.io/2016/02/12/where-to-put-std-move.html